### DISCUSSION TITLE
Feature Request: Autoplay when Bluetooth device connects

### DISCUSSION THREAD

**Discussion by andremina** [PaulWoitaschek/Voice#1369] [Ideas]
Please include the following details in your bug report:

App-Version:
Phone Name: Pixel XL
Android Version: 7.1.2

**Comment by andremina** [comment]
Best audiobook player on Android right now! 

I would love to see autoplay when phone connect to Bluetooth device. If it's an unnecessary hassle, would you be able to post the intents? So I can automate this with Tasker. Thanks!

**Comment by PaulWoitaschek** [comment]
What intents do you mean?

**Comment by andremina** [comment]
Thank you for taking the time. 

It would be what's in Step 5 here:

https://www.google.ca/amp/s/androidtrix.wordpress.com/2013/11/05/tasker-spotify-plug-load-and-play/amp/

**Comment by PaulWoitaschek** [comment]
We use the default intents:

`Intent.ACTION_MEDIA_BUTTON` with `KeyEvent.KEYCODE_MEDIA_PLAY` as extra

**Comment by PaulWoitaschek** [comment]
Does that work? You need to start `de.ph1b.audiobook.playback.PlaybackService`

**Comment by andremina** [comment]
I tried the intent with the extra but I couldn't get it to work. 

de.ph1b.audiobook.playback.PlaybackService
How do I start that? Is it a MIME type or DATA or PACKAGE or CLASS? If it's something else what is it and should it come before the intent in the sequence in Tasker?

**Comment by PaulWoitaschek** [comment]
You need to send an action of `Intent.ACTION_MEDIA_BUTTON` with the `KeyEvent.KEYCODE_MEDIA_PLAY` extra set to true to the service `de.ph1b.audiobook.playback.PlaybackService` in `de.ph1b.audiobook`.

**Comment by jessecarroll** [comment]
@andremina did you ever get this working? I'm also struggling with activating the playCurrent intent from tasker. I've tried many many variations in the settings for the action but haven't been able to trigger the intent yet. Based on what Paul said, this is my best guess so far but I can't tell what is wrong with it:

![screenshot_20171011-111718](https://user-images.githubusercontent.com/32709564/31453109-9a2d61aa-ae76-11e7-91d3-1ca529dc6c59.jpg)

**Comment by andremina** [comment]
I used an app called Bluetooth connect and play, from the App store

**Comment by holocronweaver** [comment]
Would it be possible to implement this in Voice itself? There is already the 'Resume playback' option, but I assume it only works with wired connections. I expected it to work with both bluetooth and wired.

**Comment by holocronweaver** [comment]
Ok, to support this in Voice it seems we need to implement BluetoothProfile.ServiceListener to be notified when Bluetooth status changes. That should be all that is needed to save and restore playback. Will add to my TODO.

**Comment by holocronweaver** [comment]
Alright, I implemented autoplay on Bluetooth connect. It reuses the "Resume playback" preference that was originally used for car and wired connects. 

Road tested it for a day, seems to work well. Created a related issue (and workaround) for Android auto-closing the app which breaks automatic playback resume: #709

**Comment by MiK1977** [comment]
I want to automate Voice playback using Tasker, but I have no success with the intent description above. Has anyone succeeded using Tasker to start Voice Playback?

**Comment by ryanstubbs** [comment]
If it helps anyone else, I've been having the same problem and have got it working fine in MacroDroid if you use the media action of start/pause playing and set the app as Voice. I personally use it so that if Voice is opened (which I do with Google Assistant) and my car's bluetooth is connected, it starts playing.
